File: warrior/src/main/java/com/dumbdogdiner/warrior/listeners/LobbySessionListener.java
package com.dumbdogdiner.warrior.listeners;

import com.dumbdogdiner.warrior.user.User;
import com.dumbdogdiner.warrior.api.sessions.LobbySession;
import com.dumbdogdiner.warrior.managers.PlayerManager;
import org.bukkit.GameMode;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.event.entity.EntityPickupItemEvent;
import org.bukkit.event.entity.PlayerDeathEvent;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.player.PlayerDropItemEvent;
import org.bukkit.event.player.PlayerSwapHandItemsEvent;

public class LobbySessionListener implements Listener {

    @EventHandler(priority = EventPriority.LOWEST)
    public void onDamage(EntityDamageEvent e) {
        if(!(e.getEntity() instanceof Player)) return;
        if(((Player)e.getEntity()).getGameMode().equals(GameMode.CREATIVE)) return;
        User user = PlayerManager.get(e.getEntity().getUniqueId());
        if(user == null) return;

        if(!(user.getSession() instanceof LobbySession)) return;
        e.setCancelled(true);
    }

    @EventHandler(priority = EventPriority.LOWEST)
    public void onItemDrop(PlayerDropItemEvent e) {
        if(e.getPlayer().getGameMode().equals(GameMode.CREATIVE)) return;
        User user = PlayerManager.get(e.getPlayer().getUniqueId());
        if(user == null) return;

        if(!(user.getSession() instanceof LobbySession)) return;
        e.setCancelled(true);
    }

    @EventHandler(priority = EventPriority.LOWEST)
    public void onItemSwap(PlayerSwapHandItemsEvent e) {
        if(e.getPlayer().getGameMode().equals(GameMode.CREATIVE)) return;
        User user = PlayerManager.get(e.getPlayer().getUniqueId());
        if(user == null) return;

        if(!(user.getSession() instanceof LobbySession)) return;
        e.setCancelled(true);
    }

    @EventHandler
    public void onItemPickup(EntityPickupItemEvent e) {
        if(!(e.getEntity() instanceof Player)) return;
        if(((Player)e.getEntity()).getGameMode().equals(GameMode.CREATIVE)) return;
        User user = PlayerManager.get(e.getEntity().getUniqueId());
        if(user == null) return;

        if(!(user.getSession() instanceof LobbySession)) return;
        e.setCancelled(true);
    }

    @EventHandler
    public void onDeath(PlayerDeathEvent e) {
        if(e.getEntity().getGameMode().equals(GameMode.CREATIVE)) return;
        User user = PlayerManager.get(e.getEntity().getUniqueId());
        if(user == null) return;

        if(!(user.getSession() instanceof LobbySession)) return;
        e.getDrops().clear();
    }

    @EventHandler(priority = EventPriority.LOWEST)
    public void onItemClick(InventoryClickEvent e) {
        if(e.getWhoClicked().getGameMode().equals(GameMode.CREATIVE)) return;
        User user = PlayerManager.get(e.getWhoClicked().getUniqueId());
        if(user == null) return;

        if(!(user.getSession() instanceof LobbySession)) return;
        e.setCancelled(true);
        e.getWhoClicked().closeInventory();
    }

}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/listeners/LobbySessionListener.kt
package com.dumbdogdiner.warrior.listeners

import com.dumbdogdiner.warrior.Warrior
import com.dumbdogdiner.warrior.api.sessions.LobbySession
import com.dumbdogdiner.warrior.user.User
import org.bukkit.GameMode
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.event.entity.EntityPickupItemEvent
import org.bukkit.event.entity.PlayerDeathEvent
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.player.PlayerDropItemEvent
import org.bukkit.event.player.PlayerSwapHandItemsEvent

class LobbySessionListener : Listener {

    @EventHandler(priority = EventPriority.LOWEST)
    fun onDamage(e: EntityDamageEvent) {
        if (e.entity !is Player) return
        if ((e.entity as Player).gameMode == GameMode.CREATIVE) return

        val user: User = Warrior.userCache[e.entity.uniqueId]
        if (user.session !is LobbySession) return

        e.isCancelled = true
    }

    @EventHandler(priority = EventPriority.LOWEST)
    fun onItemDrop(e: PlayerDropItemEvent) {
        if (e.player.gameMode == GameMode.CREATIVE) return

        val user: User = Warrior.userCache[e.player.uniqueId]
        if (user.session !is LobbySession) return

        e.isCancelled = true
    }

    @EventHandler(priority = EventPriority.LOWEST)
    fun onItemSwap(e: PlayerSwapHandItemsEvent) {
        if (e.player.gameMode == GameMode.CREATIVE) return

        val user: User = Warrior.userCache[e.player.uniqueId]
        if (user.session !is LobbySession) return

        e.isCancelled = true
    }

    @EventHandler
    fun onItemPickup(e: EntityPickupItemEvent) {
        if (e.entity !is Player) return
        if ((e.entity as Player).gameMode == GameMode.CREATIVE) return

        val user: User = Warrior.userCache[e.entity.uniqueId]
        if (user.session !is LobbySession) return

        e.isCancelled = true
    }

    @EventHandler
    fun onDeath(e: PlayerDeathEvent) {
        if (e.entity.gameMode == GameMode.CREATIVE) return

        val user: User = Warrior.userCache[e.entity.uniqueId]
        if (user.session !is LobbySession) return

        e.drops.clear()
    }

    @EventHandler(priority = EventPriority.LOWEST)
    fun onItemClick(e: InventoryClickEvent) {
        if (e.whoClicked.gameMode == GameMode.CREATIVE) return

        val user: User = Warrior.userCache[e.whoClicked.uniqueId]
        if (user.session !is LobbySession) return

        e.isCancelled = true
        e.whoClicked.closeInventory()
    }
}